File: Python/2485.py
# 2485. Find the Pivot Integer
# https://leetcode.com/problems/find-the-pivot-integer/

class Solution:
  def pivotInteger(self, n: int) -> int:
    sum1 = 0
    sum2 = 0
    out = -1

    for i in range(1, n + 1):
      sum1 += i

    for i in range(1, n + 1):
      sum2 += i

      if (sum1 == sum2):
        out = i
        break

      sum1 -= i

    return out

File: Rust/2485.rs
/**
 * 2485. Find the Pivot Integer
 * https://leetcode.com/problems/find-the-pivot-integer/
 */

impl Solution {
  pub fn pivot_integer(n: i32) -> i32 {
    let mut sum1: i32 = 0;
    let mut sum2: i32 = 0;
    let mut out: i32 = -1;

    for i in 1..=n {
      sum1 += i;
    }

    for i in 1..=n {
      sum2 += i;

      if sum1 == sum2 {
        out = i;
        break;
      }

      sum1 -= i;
    }

    return out;
  }
}
